sdk/core/Azure.Core/src/Shared/Page.cs

```diff
@@ -10,6 +10,9 @@ namespace Azure.Core
 {
     internal static class Page
     {
+        public static Page<T> FromValues<T>(IReadOnlyList<T> values, string? continuationToken, Response response) =>
```

Comment:
Curious - why add this rather than calling `Page<T>`.FromValues directly?

Reply:
Good question - It appears that it already has the required signature:

https://github.com/Azure/azure-sdk-for-net/blob/66a35ea758d4d70f31f77049f77ed554e3e7ce74/sdk/core/Azure.Core/src/Page.cs#L49

Reply:
It was fewer changes to call-sites, and I thought it made it easier to fall into the pit of success because the compiler would pick the best method for you.

I think (maybe longer term) the non-generic Page class wants to be got rid of It only looks to have a few references left so if you're happy with me folding that in to this change it'll probably only be about an hour.

Reply:
Just to clarify from above: It actually confuses the compiler when you're dealing with `List<T>` instances.

To call the method in `Page<T>` you need to specify the generic type, like this: 

```csharp
return Page.FromValues<TestResource>(result.Select(o => new TestResource()).ToList(), null, new MockResponse(200));
```

The compiler fades out the `TestResource` generic parameter and tells you that you can remove it, which means it will then call the IEnumerable variant on the non-generic class

Reply:
Maybe a slightly bigger task than I anticipated. There's at least 400 references to that Page class that would need updating if it was removed.

Adding a specific overload to that class would pick off the easy items (where it's already an `IReadOnlyList<T>`) rather than having to visit each call site manually, but that has a risk of introducing unexpected behaviour if someone was relying on the collection being copied.

I've removed the extra method and updated the call sites in Tables.

Reply:
Very nice update, and thank you!

Reply:
I'm on a mission to eradicate duplicating lists...! Have got rid of them in our own codebase so looking for others now :)

Happy for this to be resolved?

Reply:
Yes - if there are no changes to Core shared source files, I am happy :)  I defer to @christothes on all things Tables related.